src/overlays/actors/ovl_En_Osn/z_en_osn.c

```diff
@@ -74,59 +101,835 @@ static DamageTable D_80AD2550 = {
+    u8 temp_v0 = Message_GetState(&globalCtx->msgCtx);
+
+    if ((temp_v0 == 6 || temp_v0 == 5) && func_80147624(globalCtx)) {
+        if (this->unk_1EA & 0x20) {
+            this->unk_1EA &= ~0x20;
+            globalCtx->msgCtx.msgMode = 0x43;
+            globalCtx->msgCtx.unk12023 = 4;
+            this->actionFunc = func_80AD14C8;
+        } else {
+            func_80AD10FC(this, globalCtx);
+        }
+    }
+}
+
+void EnOsn_Idle(EnOsn* this, GlobalContext* globalCtx) {
+}
+
+void EnOsn_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnOsn* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 20.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &object_osn_Skel_0202F0, &object_osn_Anim_0201BC, 0, 0, 0);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+    this->unk_1FA = 255;
+    switch (ENOSN_GET_3(THIS)) {
+        case 0:
+            if (((gSaveContext.entranceIndex == 0xC020) || (gSaveContext.entranceIndex == 0xC030)) ||
+                (gSaveContext.entranceIndex == 0xC060)) {
+                this->unk_1EA |= 1;
+            }
+            this->unk_1F0 = 1;
+            if (globalCtx->sceneNum == SCENE_INSIDETOWER) {
+                if ((gSaveContext.entranceIndex == 0xC020) || (gSaveContext.entranceIndex == 0xC060)) {
+                    this->actionFunc = func_80AD16A8;
+                    return;
+                }
+                if (gSaveContext.entranceIndex == 0xC030) {
+                    func_80AD1398(this);
+                    this->actionFunc = func_80AD1634;
+                    return;
+                }
+                func_80AD144C(this, globalCtx);
+                return;
+            }
+            func_80AD144C(this, globalCtx);
+            break;
+
+        case 1:
+            this->unk_1EC = 0xF;
+            Actor_ChangeAnimation(&this->skelAnime, sAnimations, this->unk_1EC);
+            this->actionFunc = EnOsn_Idle;
+            break;
+
+        case 2:
+            this->unk_1EC = 0x10;
+            Actor_ChangeAnimation(&this->skelAnime, sAnimations, this->unk_1EC);
+            this->actionFunc = EnOsn_Idle;
+            break;
+
+        case 3:
+            this->actor.flags &= ~ACTOR_FLAG_1;
+            this->actionFunc = func_80AD16A8;
+            break;
+
+        default:
+            Actor_MarkForDeath(&this->actor);
+    }
+}
+
+void EnOsn_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnOsn* this = THIS;
+
+    SkelAnime_Free(&this->skelAnime, globalCtx);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnOsn_Update(Actor* thisx, GlobalContext* globalCtx) {
+    u32 pad1;
+    u16 pad2;
+    u32 sp34;
+    EnOsn* this = THIS;
```

Comment:
```suggestion
    s32 pad;
    EnOsn* this = THIS;
    u32 sp34;
```
Out of curiosity does this work?